en/animation/use-animation-curve.md

```diff
@@ -0,0 +1,322 @@
+
+When the target path is `string` or `number`, it addresses the property of the higher-level object, which itself specifies the property name. Otherwise, the target path must be an object that implements the interface `animation.TargetPath`.
+
+Cocos Creator has the following built-in classes that implement the interface `animation.TargetPath`:
+- `animation.HierarchyPath` treats a higher-level object as a node and addresses one of its children
+- `animation.ComponentPath` treats a higher-level object as a node and addresses one of its components
+
+The target paths can be any combination, as long as they have the correct meaning:
+
+```ts
+modifiers: [
+    // The target object is the first element of the "names" property of the "BlahBlahComponent" component on the "nested_3" child node of the "nested_2" child node of the "nested_1" child node
+    new animation.HierarchyPath('nested_1/nested_2/nested_3'),
+    new animation.ComponentPath(js.getClassName(BlahBlahComponent)),
+    'names',
+    0,
+]
+```
+
+Custom target paths are useful when the target object is not an property, but must be returned from a method:
+
+```ts
+class BlahBlahComponent extends Component {
+    public getName(index: number) { return _names[index]; }
+    private _names: string[] = [];
+}
+
+modifiers: [
+    // The target object is the first "name" of the "BlahBlahComponent" component on the "nested_3" child node of the "nested_2" child node of the "nested_1" child node
+    new animation.HierarchyPath('nested_1/nested_2/nested_3'),
+    new animation.ComponentPath(js.getClassName(BlahBlahComponent)),
+    {
+        get: (target: BlahBlahComponent) => target.getName(0),
+    },
+]
+```
+
+If you want custom target paths to be serializable, you can declare them as classes.
+
+```ts
+@ccclass
+class MyPath implements animation.TargetPath {
+    @property
+    public index = 0;
+    constructor(index: number) { this.index = index; }
+    get (target: BlahBlahComponent) {
+        return target.getName(this.index);
+    }
+}
+
+modifiers: [
+    // The target object is the first "name" of the "BlahBlahComponent" component on the "nested_3" child node of the "nested_2" child node of the "nested_1" child node
+    new animation.HierarchyPath('nested_1/nested_2/nested_3'),
+    new animation.ComponentPath(js.getClassName(BlahBlahComponent)),
+    new MyPath(0),
+]
+```
+
+The addressing of the target object is done at runtime, a feature that allows animation clips to be reused on multiple objects.
+
+## Assignment
+
+When a value is sampled, by default the value will be set to the target object using the assignment operator `=`.
+
+Sometimes, however, the assignment operator cannot be used to complete the setting. For example, when setting the Uniform of a material object, you can't do it with the assignment operator. For this case, the curve field `valueAdapter` provides a mechanism to customize the value to the target object.
```

Comment:
```suggestion
Sometimes, the assignment operator cannot be used to complete the setting. For example, when setting the Uniform of a material object, it can't be done with the assignment operator. For this case, the curve field `valueAdapter` provides a mechanism to customize the value to the target object.
```